### PR TITLE
fix: logout error

### PR DESCRIPTION
**Pull request by kittybest**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`yarn build`) was run locally and any changes were pushed
- [x] Lint (`yarn lint --check`) has passed locally and any fixes were made for failures

## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?

- fix web crash after logout
- set both `userContext.id` and `userContext.userState` to `undefined` after logout

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/frontend/src/context/User.ts [hunk 1]
```diff
@@ -92,6 +92,8 @@ export class User {
     save() {
         if (this.id) {
             window.localStorage.setItem('identity', this.id.serializeIdentity())
+        } else {
+            window.localStorage.removeItem('identity')
```
Comment (vimwitch) [inline packages/frontend/src/context/User.ts:96]: We also need to wipe the indexed DB storage.

File: packages/frontend/cypress/e2e/signup.cy.ts [hunk 2]
```diff
@@ -75,7 +75,6 @@ describe('visit and interact with home page', () => {
         }).as('getApiComment')
         cy.get('#new > img').click()
         cy.location('pathname').should('eq', '/new')
-        cy.get('.rep-info').contains('30')
```
Comment (vimwitch) [inline packages/frontend/cypress/e2e/signup.cy.ts:78]: Not sure why this line was causing test failures, but we definitely shouldn't remove it.

**Review by vimwitch — CHANGES_REQUESTED** [review]

File: packages/frontend/src/context/User.ts (hunk 1 above)
Comment (kittybest) [reply packages/frontend/src/context/User.ts:96]: So should I use `db.close()`?

File: packages/frontend/cypress/e2e/signup.cy.ts (hunk 2 above)
Comment (kittybest) [reply packages/frontend/cypress/e2e/signup.cy.ts:78]: after I wipe out the indexedDB, the test passed

File: packages/frontend/src/context/User.ts (hunk 1 above)
Comment (vimwitch) [reply packages/frontend/src/context/User.ts:98]: ah `db.close()` doesn't change the database, it just closes the handle. You need to use `db.delete('GSTLeaf', { where: {} })` for each collection

**Comment by vimwitch** [comment]
This fix is included in #95

**Comment by kittybest** [comment]
> This fix is included in #95

you're right, I've tested it, closed.

**Pull request closed without merging**